app/controllers/parameters_controller.rb

```diff
@@ -1,16 +1,38 @@
 class ParametersController < ApplicationController
-  skip_before_action :authenticate_user!, only: [:new, :create]
+  skip_before_action :authenticate_user!, only: [:new, :create, :show]
   before_action :set_params, only: [:create]
 
   def show
+    FakeData.where('booked != true').destroy_all
+    Parameter.where('id != ?', params[:id]).destroy_all
     # FakeData.destroy_all
     @booking = Booking.new
     @parameter = Parameter.find(params[:id])
     FakeData.generate_results(@parameter)
-    @fastest = FakeData.all.min_by(&:duration)
-    @cheapest = FakeData.all.min_by(&:cost)
 
-    @recommended = FakeData.find(recommended(FakeData.all, @parameter.preferred_start).first)
+    valid_data = FakeData.where("cost != 0 AND booked != true AND end_time < ? AND start_time > ?", @parameter.latest_finish, @parameter.earliest_start)
```

Comment:
Minor thing but this should be <= as we want to include the last time they specify as an option